File: common/src/models/dashboard_item.rs
use chrono::DateTime;
use serde::{Serialize, Deserialize};

#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct DashboardItem {
    pub room_id: i64,
    pub room_name: String,
    pub last_time: DateTime<chrono::Utc>,
    pub temperature: f64,
    pub humidity: f64,
    pub avg_duration_seconds: i64,
    pub avg_temperature: f64,
    pub avg_humidity: f64,
}

File: frontend/types/dashboardItem.ts
export class DashboardItemData {
    room_id: number;
    room_name: string;
    last_time: string;
    temperature: number;
    humidity: number;
    avg_duration_seconds: number;
    avg_temperature: number;
    avg_humidity: number;

    constructor(room_id: number, room_name: string, last_time: string, temperature: number, humidity: number, avg_duration_seconds: number, avg_temperature: number, avg_humidity: number) {
        this.room_id = room_id;
        this.room_name = room_name;
        this.last_time = last_time;
        this.temperature = temperature;
        this.humidity = humidity;
        this.avg_duration_seconds = avg_duration_seconds;
        this.avg_temperature = avg_temperature;
        this.avg_humidity = avg_humidity;
    }
}
